<!DOCTYPE html>
<!-- saved from url=(0022)http://localhost:4845/ -->
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">


<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Evan">

<title>MBMI 5030_Final Project</title>
<style class="anchorjs"></style><style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="./MBMI 5030_Final Project_files/clipboard.min.js.download"></script>
<script src="./MBMI 5030_Final Project_files/quarto.js.download"></script>
<script src="./MBMI 5030_Final Project_files/popper.min.js.download"></script>
<script src="./MBMI 5030_Final Project_files/tippy.umd.min.js.download"></script>
<script src="./MBMI 5030_Final Project_files/anchor.min.js.download"></script>
<link href="./MBMI 5030_Final Project_files/tippy.css" rel="stylesheet">
<link href="./MBMI 5030_Final Project_files/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="./MBMI 5030_Final Project_files/bootstrap.min.js.download"></script>
<link href="./MBMI 5030_Final Project_files/bootstrap-icons.css" rel="stylesheet">
<link href="./MBMI 5030_Final Project_files/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


<style>.ansi-display{font-family:monospace;white-space:pre-wrap;--ansiBlack: #000000;--ansiRed: #cd3131;--ansiGreen: #00BC00;--ansiYellow: #949800;--ansiBlue: #0451a5;--ansiMagenta: #bc05bc;--ansiCyan: #0598bc;--ansiWhite: #555555;--ansiBrightBlack: #666666;--ansiBrightRed: #cd3131;--ansiBrightGreen: #14CE14;--ansiBrightYellow: #b5ba00;--ansiBrightBlue: #0451a5;--ansiBrightMagenta: #bc05bc;--ansiBrightCyan: #0598bc;--ansiBrightWhite: #a5a5a5}.dark-mode .ansi-display{--ansiBlack: #000000;--ansiRed: #cd3131;--ansiGreen: #0DBC79;--ansiYellow: #e5e510;--ansiBlue: #2472c8;--ansiMagenta: #bc3fbc;--ansiCyan: #11a8cd;--ansiWhite: #e5e5e5;--ansiBrightBlack: #666666;--ansiBrightRed: #f14c4c;--ansiBrightGreen: #23d18b;--ansiBrightYellow: #f5f543;--ansiBrightBlue: #3b8eea;--ansiBrightMagenta: #d670d6;--ansiBrightCyan: #29b8db;--ansiBrightWhite: #e5e5e5}@keyframes ansi-displaly-run-blink{50%{opacity:0}}.fui-FluentProvider h2{border-bottom:none;padding-bottom:0}</style><style data-make-styles-bucket="d" data-priority="0"></style><style id="fui-FluentProviderr0"></style></head>

<body class="fullcontent quarto-light" data-tabster="{&quot;root&quot;:{}}">

<i tabindex="0" role="none" data-tabster-dummy="" aria-hidden="true" style="position: fixed; height: 1px; width: 1px; opacity: 0.001; z-index: -1; content-visibility: hidden; top: 0px; left: 0px;"></i><div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">MBMI 5030_Final Project</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Evan </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="overview" class="level1">
<h1>Overview</h1>
<p>Prostate cancer is the second most common type of cancer in the United States, with an estimated 300,000 new cases in 2024. Despite recent innovations (e.g., radioligand therapeutics, PMSA Imaging), there remains distinct patient-segments with high unmet need due to unique tumor characteristics. Disease aggressiveness, a term that is often ambiguously defined and inconsistently treated in real-world setting, refers to an aggregate blend of patient features that includes prognostic markers (Gleason Score), altered genetic markers (P53, RB, MYC), rapid disease progression, and presence of visceral metastases (cancer that has spread to organs). Within this analysis my goal is to create a clinically actionable definition of ‘aggressive disease’ by integrating prognostic variables to form a compositive score. I will then evaluate whether we can use the presence of a genetic mutation to predict whether a patient will classify as having aggressive disease.</p>
</section>
<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>The prostate cancer diagnostic and treatment armamentarium is variable and at times highly complex due evolving US diagnostic guidelines and distinctive treatment patterns amongst HCPs (urologist vs.&nbsp;medical oncologist). Prostate-specific antigen (PSA) testing is a blood test that helps to detect likelihood of disease via rising PSA levels. Despite broad adoption, the effectiveness of PSA as a screening modality is debated, with multiple guidelines changes in recent decades.</p>
<ul>
<li><p>Prior to 2008, PSA testing was recommended for all men &gt;50.</p></li>
<li><p>However, studies demonstrated that ~50 men had to be treated to save one life. The burden of treatment was seen as unnecessarily high, leading to discontinuation of routine testing.</p></li>
<li><p>With less PSA testing the diagnosis rate of metastatic disease increased. This resulted in a re-evaluation in 2017, with present literature suggesting that men with risk factors such as BRCA mutated, specific minority status, or immediate family histories discuss screening at age 40 – 45.</p></li>
</ul>
<p>The Gleason Score (GS) is a pathological grading system used to determine prognosis and to help inform optimal treatment approach. The higher the Gleason Score, the more abnormal the tissue and more likely the cancer will grow quickly. GS is also used in conjunction with clinical stage, serum PSA, and tumor volume to help determine treatment for newly diagnosed patients.</p>
<ul>
<li>The GS score is calculated by rating differentiation amongst the 2 most prevalent tumor patterns on a 1-5 scale, and then summing to find a total composite score out of 10. A GS of 2 – 6 is considered low risk, 7 is considered intermediate risk, and 8 – 10 is considered high risk.</li>
</ul>
<p>This brings us back to the phrase aggressive disease. Not only do patients with aggressive disease have meaningfully worse health outcomes, but approach to treatment is reliant on early identification and HCP treatment sequencing decisions in prior lines of therapy. Lack of a standard definition further confounds this challenge (e.g., The LATTITUDE and CHAARTED trials used different criteria to define high-risk disease). To address this challenge and improve patient outcomes we must develop a more cohesive definition for aggressive disease and identify clinical variables that can predict aggressiveness. In this analysis I will examine the significance of TMRPSS2-ERG fusion status and determine whether there is predictive capability for disease aggressiveness.</p>
</section>
<section id="r-packages" class="level1">
<h1>R Packages</h1>
<p>The following packages were read into this project to support data standardization and analysis.</p>
<p>The <code>echo: false</code> option disables the printing of code (only output is displayed).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="http://localhost:4845/#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="http://localhost:4845/#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"naniar"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'naniar' was built under R version 4.4.2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="http://localhost:4845/#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"mice"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'mice' was built under R version 4.4.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'mice'

The following object is masked from 'package:stats':

    filter

The following objects are masked from 'package:base':

    cbind, rbind</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="http://localhost:4845/#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"factoextra"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'factoextra' was built under R version 4.4.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome! Want to learn more? See two factoextra-related books at https://goo.gl/ve3WBa</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="http://localhost:4845/#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'magrittr'

The following object is masked from 'package:purrr':

    set_names

The following object is masked from 'package:tidyr':

    extract</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="http://localhost:4845/#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb13-2"><a href="http://localhost:4845/#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(caret)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'caret' was built under R version 4.4.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: lattice

Attaching package: 'caret'

The following object is masked from 'package:purrr':

    lift</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="http://localhost:4845/#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(randomForest)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>randomForest 4.7-1.2
Type rfNews() to see new features/changes/bug fixes.

Attaching package: 'randomForest'

The following object is masked from 'package:dplyr':

    combine

The following object is masked from 'package:ggplot2':

    margin</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="http://localhost:4845/#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmnet)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'glmnet' was built under R version 4.4.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Matrix

Attaching package: 'Matrix'

The following objects are masked from 'package:tidyr':

    expand, pack, unpack

Loaded glmnet 4.1-8</code></pre>
</div>
</div>
</section>
<section id="methodology" class="level1">
<h1>Methodology</h1>
<p>The dataset that I utilized in this project is from the cBioPortal for Cancer Genomics. The cBioPortal is a well-established open-source platform that allows researchers to engage in interactive exploration of cancer genomics data sets (data is curated from &gt;=350 studies). The data of interest, Prostate Adenocarcinoma (Broad/Cornell, Nat Genet 2012), was generated by Levi Garraway’s lab at the Broad Institute and Mark Rubin’s lab at Cornell. I was able to evaluate and perform supervised machine learning analyses on the N=123 unique patient samples due to presence of a case/control, which took the form of TMPRSS2-ERG Fusion Status – with ~50% of the patients positive, ~50% negative. To initiate the analysis I had to import the dataset from the cBioPortal. I loaded library(“table1”) to help create formatted summary tables for the clinical data, which was obtained from cBioPortal in TSV format. From there I used name_repair = “universal_quiet” for naming standardization and syntax alignment. Lastly, I used the rename function to change the name of some of the lengthier column names to make it easier to analyze the data.</p>
<p>There are n=123 rows read in, which represents N=123 unique patient samples from the cBioPortal data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="http://localhost:4845/#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"table1"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'table1' was built under R version 4.4.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'table1'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    units, units&lt;-</code></pre>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="http://localhost:4845/#cb25-1" aria-hidden="true" tabindex="-1"></a>Clinical<span class="ot">=</span><span class="fu">read_tsv</span>(<span class="st">"prad_broad_clinical_data (2).tsv"</span>, <span class="at">name_repair =</span> <span class="st">"universal_quiet"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 123 Columns: 19</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (11): Study.ID, Patient.ID, Sample.ID, Cancer.Type, Cancer.Type.Detailed...
dbl  (8): Diagnosis.Age, Fraction.Genome.Altered, Gleason.Score.Percent.4......

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="http://localhost:4845/#cb28-1" aria-hidden="true" tabindex="-1"></a>Clinical <span class="ot">=</span> Clinical <span class="sc">|&gt;</span></span>
<span id="cb28-2"><a href="http://localhost:4845/#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Age =</span> Diagnosis.Age, </span>
<span id="cb28-3"><a href="http://localhost:4845/#cb28-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">FGA.Status =</span> Fraction.Genome.Altered,</span>
<span id="cb28-4"><a href="http://localhost:4845/#cb28-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">Prostatectomy.Gleason.Score =</span> Radical.Prostatectomy.Gleason.Score.for.Prostate.Cancer,</span>
<span id="cb28-5"><a href="http://localhost:4845/#cb28-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">Capra.Score =</span> Modified.Capra.S.Score.Without.Including.Surgical.Margins,</span>
<span id="cb28-6"><a href="http://localhost:4845/#cb28-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">PSA.Status =</span> Serum.PSA,</span>
<span id="cb28-7"><a href="http://localhost:4845/#cb28-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">Tumor.Mutation.Burden =</span> TMB..nonsynonymous.)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="assessed-missingness" class="level2">
<h2 class="anchored" data-anchor-id="assessed-missingness">Assessed Missingness<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#assessed-missingness" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<p>My next The functionvis_miss was used to determine that the dataset had several missing values.</p>
<ul>
<li><p>Rows with missing ‘age’ inputs were excluded from analysis.</p></li>
<li><p>In situations where FGA status was absent the <code>mice</code> library was used to impute the data. FGA was imputed given this variable was identified as a key element for the composite score used to predict aggressiveness of disease.</p></li>
<li><p>Out of the N=123 original samples, N=112 were ultimately included in the final analysis.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="http://localhost:4845/#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Assess Missingness</span></span>
<span id="cb29-2"><a href="http://localhost:4845/#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="fu">vis_miss</span>(Clinical)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="http://localhost:4845/#cb30-1" aria-hidden="true" tabindex="-1"></a>Clinical <span class="ot">=</span> Clinical <span class="sc">|&gt;</span></span>
<span id="cb30-2"><a href="http://localhost:4845/#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">drop_na</span>(Age)</span>
<span id="cb30-3"><a href="http://localhost:4845/#cb30-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-4"><a href="http://localhost:4845/#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="fu">vis_miss</span>(Clinical)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-3-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="http://localhost:4845/#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Going to impute FGA where we have some missingness </span></span>
<span id="cb31-2"><a href="http://localhost:4845/#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Installed MICE Library</span></span>
<span id="cb31-3"><a href="http://localhost:4845/#cb31-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-4"><a href="http://localhost:4845/#cb31-4" aria-hidden="true" tabindex="-1"></a>Clinical.mids <span class="ot">=</span> <span class="fu">mice</span>(Clinical)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 iter imp variable
  1   1  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  1   2  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  1   3  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  1   4  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  1   5  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  2   1  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  2   2  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  2   3  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  2   4  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  2   5  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  3   1  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  3   2  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  3   3  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  3   4  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  3   5  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  4   1  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  4   2  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  4   3  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  4   4  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  4   5  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  5   1  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  5   2  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  5   3  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  5   4  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status
  5   5  FGA.Status  Gleason.Score.Percent.4...5  Capra.Score  PSA.Status</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Number of logged events: 13</code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="http://localhost:4845/#cb34-1" aria-hidden="true" tabindex="-1"></a>Clinical <span class="ot">=</span> <span class="fu">complete</span>(Clinical.mids) <span class="do">## Gives clinical dataframe with 112 obs &amp; 0 missing</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>New names:
• `Gleason.Score.Percent.4...5` -&gt; `Gleason.Score.Percent.4`</code></pre>
</div>
</div>
</section>
<section id="outlier-removal-new-column" class="level2">
<h2 class="anchored" data-anchor-id="outlier-removal-new-column">Outlier Removal, New Column<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#outlier-removal-new-column" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li><p>A single score for tumor mutation burden was removed given representation as an extreme outlier</p></li>
<li><p>FGA.Status was created as a new column to support data normalization and ease of working with any data variability</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="http://localhost:4845/#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Exclude N=1 TMB outlier</span></span>
<span id="cb36-2"><a href="http://localhost:4845/#cb36-2" aria-hidden="true" tabindex="-1"></a>Clinical <span class="ot">=</span> Clinical <span class="sc">|&gt;</span></span>
<span id="cb36-3"><a href="http://localhost:4845/#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Tumor.Mutation.Burden<span class="sc">&lt;=</span><span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb36-4"><a href="http://localhost:4845/#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Log.FGA.Status=</span><span class="fu">log</span>(FGA.Status)) </span>
<span id="cb36-5"><a href="http://localhost:4845/#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="do">## New FGA Column with Imputed Columns</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="simplified-representation-of-tmprss2.erg.fusion.status" class="level2">
<h2 class="anchored" data-anchor-id="simplified-representation-of-tmprss2.erg.fusion.status">Simplified Representation of TMPRSS2.ERG.Fusion.Status<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#simplified-representation-of-tmprss2.erg.fusion.status" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li><p>TMPRSS2.ERG.Fusion.Status is represented three ways within the dataset sourced from cBioPortal</p>
<ul>
<li><p>Negative</p></li>
<li><p>Positive</p></li>
<li><p>Positive with interstitial deletion</p></li>
</ul></li>
<li><p>For simplicity purposes in the upfront analysis I changed TMPRSS2.ERG.Fusion.Status to positive or negative, effectively folding the positive with interstitial deletion into the positive cohort.</p></li>
<li><p>This allows us to nearly have a 50/50 split between patients with a negative vs.&nbsp;positive mutation status.</p>
<ul>
<li>In subsequent analyses I brought back the three distinct groups for more nuanced insights and preferred predictors.</li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="http://localhost:4845/#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(Clinical<span class="sc">$</span>TMPRSS2.ERG.Fusion.Status)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
                           Negative                            Positive 
                                 53                                  24 
Positive with interstitial deletion 
                                 34 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="http://localhost:4845/#cb39-1" aria-hidden="true" tabindex="-1"></a>Clinical<span class="ot">=</span>Clinical <span class="sc">|&gt;</span></span>
<span id="cb39-2"><a href="http://localhost:4845/#cb39-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">TMPRSS2.Simplified=</span><span class="fu">factor</span>(<span class="fu">if_else</span>(TMPRSS2.ERG.Fusion.Status<span class="sc">==</span><span class="st">"Positive with interstitial deletion"</span>, </span>
<span id="cb39-3"><a href="http://localhost:4845/#cb39-3" aria-hidden="true" tabindex="-1"></a>                                           <span class="st">"Positive"</span>, TMPRSS2.ERG.Fusion.Status),</span>
<span id="cb39-4"><a href="http://localhost:4845/#cb39-4" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'Negative'</span>, <span class="st">'Positive'</span>)),</span>
<span id="cb39-5"><a href="http://localhost:4845/#cb39-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">Interstitial.Deletion =</span> <span class="fu">factor</span>(<span class="fu">if_else</span>(TMPRSS2.ERG.Fusion.Status<span class="sc">==</span><span class="st">"Positive with interstitial deletion"</span>,</span>
<span id="cb39-6"><a href="http://localhost:4845/#cb39-6" aria-hidden="true" tabindex="-1"></a>                                                <span class="st">"Yes"</span>, <span class="st">"No"</span>),</span>
<span id="cb39-7"><a href="http://localhost:4845/#cb39-7" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'No'</span>, <span class="st">'Yes'</span>)),</span>
<span id="cb39-8"><a href="http://localhost:4845/#cb39-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">TMPRSS2.Full =</span> <span class="fu">factor</span>(<span class="fu">if_else</span>(TMPRSS2.ERG.Fusion.Status<span class="sc">==</span><span class="st">"Positive"</span>, </span>
<span id="cb39-9"><a href="http://localhost:4845/#cb39-9" aria-hidden="true" tabindex="-1"></a>                                       <span class="st">"Positive, no interstitial deletion"</span>, TMPRSS2.ERG.Fusion.Status),</span>
<span id="cb39-10"><a href="http://localhost:4845/#cb39-10" aria-hidden="true" tabindex="-1"></a>                               <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">'Negative'</span>,</span>
<span id="cb39-11"><a href="http://localhost:4845/#cb39-11" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">'Positive, no interstitial deletion'</span>,</span>
<span id="cb39-12"><a href="http://localhost:4845/#cb39-12" aria-hidden="true" tabindex="-1"></a>                                          <span class="st">'Positive with interstitial deletion'</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="aggregated-radical-prostatectomy-gleason" class="level2">
<h2 class="anchored" data-anchor-id="aggregated-radical-prostatectomy-gleason">Aggregated Radical Prostatectomy Gleason<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#aggregated-radical-prostatectomy-gleason" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li><p>Combined the inputs from Radical Prostatectomy Gleason Score for Prostate Cancer</p>
<ul>
<li><p>This combined the two separate numerics and made a single composite score that was easier to compute (e.g., 3 + 4 = 7)</p></li>
<li><p>The numerics are referred to as the primary and secondary cancer cell pattern scores</p>
<ul>
<li><p>Primary grade: The grade of the largest area of the tumor</p></li>
<li><p>Secondary grade: The grade of the second largest area of the tumor</p></li>
</ul></li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="http://localhost:4845/#cb40-1" aria-hidden="true" tabindex="-1"></a>Clinical <span class="ot">=</span> Clinical <span class="sc">|&gt;</span></span>
<span id="cb40-2"><a href="http://localhost:4845/#cb40-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Tumor.Stage.Binary=</span><span class="fu">str_sub</span>(Tumor.Stage,<span class="dv">1</span>,<span class="dv">3</span>)) <span class="sc">|&gt;</span></span>
<span id="cb40-3"><a href="http://localhost:4845/#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">|&gt;</span></span>
<span id="cb40-4"><a href="http://localhost:4845/#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Composite.Gleason.Score =</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text=</span>Prostatectomy.Gleason.Score))) <span class="sc">|&gt;</span></span>
<span id="cb40-5"><a href="http://localhost:4845/#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="patient-demographics-initial-data-visualization-and-exploratory-analysis" class="level2">
<h2 class="anchored" data-anchor-id="patient-demographics-initial-data-visualization-and-exploratory-analysis">Patient Demographics: Initial Data Visualization and Exploratory Analysis<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#patient-demographics-initial-data-visualization-and-exploratory-analysis" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li>Demographic overview from available data e.g., age, tumor stage, mutation counts, mutation status, and Capra Score</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="http://localhost:4845/#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(Clinical,<span class="fu">aes</span>(<span class="at">x=</span>Age, <span class="at">fill=</span>Tumor.Stage.Binary)) <span class="sc">+</span> </span>
<span id="cb41-2"><a href="http://localhost:4845/#cb41-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">position =</span> <span class="st">"dodge"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="http://localhost:4845/#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(Clinical<span class="sc">$</span>Tumor.Stage)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
  pT2a   pT2b   pT2c   pT3a   pT3b pT3bN1 
     4      1     39     49     17      1 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="http://localhost:4845/#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Density Curve</span></span>
<span id="cb45-2"><a href="http://localhost:4845/#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(Clinical,<span class="fu">aes</span>(<span class="at">x=</span>Age, <span class="at">color=</span>Tumor.Stage.Binary)) <span class="sc">+</span> </span>
<span id="cb45-3"><a href="http://localhost:4845/#cb45-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_density</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-7-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="http://localhost:4845/#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Distribution of patients by age</span></span>
<span id="cb46-2"><a href="http://localhost:4845/#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(Clinical, <span class="fu">aes</span>(<span class="at">x =</span> Age)) <span class="sc">+</span></span>
<span id="cb46-3"><a href="http://localhost:4845/#cb46-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">5</span>, <span class="at">fill =</span> <span class="st">"blue"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb46-4"><a href="http://localhost:4845/#cb46-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Distribution of Patients by Age"</span>, <span class="at">x =</span> <span class="st">"Age"</span>, <span class="at">y =</span> <span class="st">"Frequency"</span>) <span class="sc">+</span></span>
<span id="cb46-5"><a href="http://localhost:4845/#cb46-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-8-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="http://localhost:4845/#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Distribution of patients by tumor stage</span></span>
<span id="cb47-2"><a href="http://localhost:4845/#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(Clinical, <span class="fu">aes</span>(<span class="at">x =</span> Tumor.Stage)) <span class="sc">+</span></span>
<span id="cb47-3"><a href="http://localhost:4845/#cb47-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">fill =</span> <span class="st">"orange"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb47-4"><a href="http://localhost:4845/#cb47-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Distribution of Patients by Tumor Stage"</span>, <span class="at">x =</span> <span class="st">"Tumor Stage"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb47-5"><a href="http://localhost:4845/#cb47-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-8-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="http://localhost:4845/#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Distribution of patients by mutation count </span></span>
<span id="cb48-2"><a href="http://localhost:4845/#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(Clinical, <span class="fu">aes</span>(<span class="at">x =</span> Mutation.Count)) <span class="sc">+</span></span>
<span id="cb48-3"><a href="http://localhost:4845/#cb48-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">2</span>, <span class="at">fill =</span> <span class="st">"purple"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb48-4"><a href="http://localhost:4845/#cb48-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Distribution of Patients by Mutation Count"</span>, <span class="at">x =</span> <span class="st">"Mutation Count (TMB)"</span>, <span class="at">y =</span> <span class="st">"Frequency"</span>) <span class="sc">+</span></span>
<span id="cb48-5"><a href="http://localhost:4845/#cb48-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-8-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="http://localhost:4845/#cb49-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 4. Distribution of patients with TMPRSS2-ERG fusion status</span></span>
<span id="cb49-2"><a href="http://localhost:4845/#cb49-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(Clinical, <span class="fu">aes</span>(<span class="at">x =</span> TMPRSS2.Full)) <span class="sc">+</span></span>
<span id="cb49-3"><a href="http://localhost:4845/#cb49-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">fill =</span> <span class="st">"green"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb49-4"><a href="http://localhost:4845/#cb49-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Distribution of Patients by TMPRSS2-ERG Fusion Status"</span>, <span class="at">x =</span> <span class="st">"Fusion Status"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb49-5"><a href="http://localhost:4845/#cb49-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-8-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="http://localhost:4845/#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 5. Distribution of patients with Capra Score</span></span>
<span id="cb50-2"><a href="http://localhost:4845/#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(Clinical, <span class="fu">aes</span>(<span class="at">x =</span> Capra.Score)) <span class="sc">+</span></span>
<span id="cb50-3"><a href="http://localhost:4845/#cb50-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">fill =</span> <span class="st">"brown"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb50-4"><a href="http://localhost:4845/#cb50-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Distribution of Patients by Capra Score"</span>, <span class="at">x =</span> <span class="st">"Capra Score"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb50-5"><a href="http://localhost:4845/#cb50-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-8-5.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="http://localhost:4845/#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Age by PSA Status Distribution</span></span>
<span id="cb51-2"><a href="http://localhost:4845/#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(Clinical, <span class="fu">aes</span>(<span class="at">x =</span> Age, <span class="at">y =</span> PSA.Status, <span class="at">color=</span>Tumor.Stage.Binary)) <span class="sc">+</span></span>
<span id="cb51-3"><a href="http://localhost:4845/#cb51-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb51-4"><a href="http://localhost:4845/#cb51-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Distribution of Patients by Age and PSA Stage"</span>,</span>
<span id="cb51-5"><a href="http://localhost:4845/#cb51-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Age"</span>,</span>
<span id="cb51-6"><a href="http://localhost:4845/#cb51-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"PSA Status"</span>,</span>
<span id="cb51-7"><a href="http://localhost:4845/#cb51-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">'Tumor Stage'</span>) <span class="sc">+</span></span>
<span id="cb51-8"><a href="http://localhost:4845/#cb51-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="http://localhost:4845/#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Age by Tumor Stage Distribution</span></span>
<span id="cb52-2"><a href="http://localhost:4845/#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(Clinical, <span class="fu">aes</span>(<span class="at">x =</span> Age, <span class="at">fill =</span> Tumor.Stage)) <span class="sc">+</span></span>
<span id="cb52-3"><a href="http://localhost:4845/#cb52-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="dv">5</span>, <span class="at">position =</span> <span class="st">"dodge"</span>, <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb52-4"><a href="http://localhost:4845/#cb52-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Distribution of Patients by Age and Tumor Stage"</span>,</span>
<span id="cb52-5"><a href="http://localhost:4845/#cb52-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Age"</span>,</span>
<span id="cb52-6"><a href="http://localhost:4845/#cb52-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Count of Patients"</span>) <span class="sc">+</span></span>
<span id="cb52-7"><a href="http://localhost:4845/#cb52-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_brewer</span>(<span class="at">palette =</span> <span class="st">"Set3"</span>, <span class="at">name =</span> <span class="st">"Tumor Stage"</span>) <span class="sc">+</span></span>
<span id="cb52-8"><a href="http://localhost:4845/#cb52-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="composite-score-to-define-aggressive-disease" class="level2">
<h2 class="anchored" data-anchor-id="composite-score-to-define-aggressive-disease">Composite Score to Define Aggressive Disease<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#composite-score-to-define-aggressive-disease" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li><p><strong>Three variables were combined to create a composite score that represented “aggressive disease”</strong></p>
<ul>
<li>The higher the score, the more aggressive the disease</li>
</ul></li>
<li><p><strong>Variables include:</strong></p>
<ul>
<li><p>Radical Prostatectomy Gleason Score: The sum of the two most common Gleason patterns in the tumor.&nbsp;</p></li>
<li><p>Tumor Mutation Burden (TMB): Measurement of the number of mutations in a cancer cell’s DNA.</p></li>
<li><p>Fraction Genome Altered: Measurement of the percentage of a tumor’s chromosomes that have been altered in copy number.&nbsp;</p></li>
</ul></li>
<li><p><strong>Performed model splitting (80/20)</strong></p>
<ul>
<li>The split ensures the model(s) are trained on one subset of the data and evaluated on another. Intended to prevent over-fitting.</li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="http://localhost:4845/#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Load GGally, provide extended functionality for ggplot2,</span></span>
<span id="cb53-2"><a href="http://localhost:4845/#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">"GGally"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'GGally' was built under R version 4.4.2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Registered S3 method overwritten by 'GGally':
  method from   
  +.gg   ggplot2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="http://localhost:4845/#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Pairwise plot matrix of the three specified variables</span></span>
<span id="cb56-2"><a href="http://localhost:4845/#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggpairs</span>(Clinical,<span class="at">columns=</span><span class="fu">c</span>(<span class="st">"Composite.Gleason.Score"</span>,<span class="st">"FGA.Status"</span>,<span class="st">"Tumor.Mutation.Burden"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="http://localhost:4845/#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Composite score with updated FGA to reflect imputed values</span></span>
<span id="cb57-2"><a href="http://localhost:4845/#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggpairs</span>(Clinical,<span class="at">columns=</span><span class="fu">c</span>(<span class="st">"Composite.Gleason.Score"</span>,<span class="st">"Log.FGA.Status"</span>,<span class="st">"Tumor.Mutation.Burden"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="http://localhost:4845/#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Performing splitting to support accuracy of model - 80/20 split</span></span>
<span id="cb58-2"><a href="http://localhost:4845/#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Obtaining testing dataset on which we can perform predictions and determine prediction accuracy</span></span>
<span id="cb58-3"><a href="http://localhost:4845/#cb58-3" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb58-4"><a href="http://localhost:4845/#cb58-4" aria-hidden="true" tabindex="-1"></a>clincal_Filtered<span class="ot">=</span>Clinical <span class="sc">|&gt;</span> <span class="fu">filter</span>(Mutation.Count<span class="sc">&lt;</span><span class="dv">100</span>)</span>
<span id="cb58-5"><a href="http://localhost:4845/#cb58-5" aria-hidden="true" tabindex="-1"></a>train.rows <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(Clinical), <span class="at">size=</span><span class="fu">round</span>(<span class="fl">0.8</span><span class="sc">*</span><span class="fu">nrow</span>(Clinical))))</span>
<span id="cb58-6"><a href="http://localhost:4845/#cb58-6" aria-hidden="true" tabindex="-1"></a>Clinical_Train<span class="ot">=</span>Clinical[train.rows,]</span>
<span id="cb58-7"><a href="http://localhost:4845/#cb58-7" aria-hidden="true" tabindex="-1"></a>Clinical_Test<span class="ot">=</span>Clinical[<span class="sc">-</span>train.rows,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="principal-component-analysis-pca" class="level2">
<h2 class="anchored" data-anchor-id="principal-component-analysis-pca">Principal Component Analysis (PCA)<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#principal-component-analysis-pca" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li><p>PCA Goal: Determine how Composite.Gleason.Score, Log.FGA.Status, Tumor.Mutation.Burden clinical variables are related.</p>
<ul>
<li>Identify variable combinations for use in regression analyses.</li>
</ul></li>
<li><p>PC1 explains 52.03% of the variance, PC2 explains 28.63%, and PC3 explains 19.34%.</p>
<ul>
<li><p>PC1: All three variables contribute a near equal amount (all negative #s)</p></li>
<li><p>PC2: Dominated by Gleason Score</p></li>
<li><p>PC3: Reflects the opposing relationship between Log.FGA.Status and Tumor.Mutation.Burden</p></li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="http://localhost:4845/#cb59-1" aria-hidden="true" tabindex="-1"></a>PCA_Results<span class="ot">=</span><span class="fu">prcomp</span>(<span class="sc">~</span>Composite.Gleason.Score <span class="sc">+</span> Log.FGA.Status <span class="sc">+</span> Tumor.Mutation.Burden,</span>
<span id="cb59-2"><a href="http://localhost:4845/#cb59-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">data =</span> Clinical_Train,</span>
<span id="cb59-3"><a href="http://localhost:4845/#cb59-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">scale.=</span><span class="cn">TRUE</span>)</span>
<span id="cb59-4"><a href="http://localhost:4845/#cb59-4" aria-hidden="true" tabindex="-1"></a>PCA_Results</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Standard deviations (1, .., p=3):
[1] 1.2546551 0.9293815 0.7497270

Rotation (n x k) = (3 x 3):
                               PC1        PC2         PC3
Composite.Gleason.Score -0.4409235  0.8940972 -0.07859177
Log.FGA.Status          -0.6261540 -0.3691552 -0.68677188
Tumor.Mutation.Burden   -0.6430534 -0.2536033  0.72261173</code></pre>
</div>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="http://localhost:4845/#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="do">##Shows how PCA relates to different variables</span></span>
<span id="cb61-2"><a href="http://localhost:4845/#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="fu">fviz_pca_var</span>(PCA_Results)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="bootstrapping" class="level2">
<h2 class="anchored" data-anchor-id="bootstrapping">Bootstrapping<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#bootstrapping" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li><p>Assess stability and variability in PCA score - focus on Clinical_Train (training data)</p></li>
<li><p>Replicate 5000 times</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="http://localhost:4845/#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Bootstrapping</span></span>
<span id="cb62-2"><a href="http://localhost:4845/#cb62-2" aria-hidden="true" tabindex="-1"></a>Get_Principal_Component_One<span class="ot">=</span><span class="cf">function</span>(data){</span>
<span id="cb62-3"><a href="http://localhost:4845/#cb62-3" aria-hidden="true" tabindex="-1"></a>bootstrapped_rows<span class="ot">=</span><span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(data),<span class="at">replace=</span><span class="cn">TRUE</span>)</span>
<span id="cb62-4"><a href="http://localhost:4845/#cb62-4" aria-hidden="true" tabindex="-1"></a>  PCA_Results<span class="ot">=</span><span class="fu">prcomp</span>(<span class="sc">~</span>Composite.Gleason.Score <span class="sc">+</span> Log.FGA.Status <span class="sc">+</span> Tumor.Mutation.Burden,</span>
<span id="cb62-5"><a href="http://localhost:4845/#cb62-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">data =</span> data[bootstrapped_rows, ], <span class="do">## Doing PCA with bootstrapped sample </span></span>
<span id="cb62-6"><a href="http://localhost:4845/#cb62-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">scale.=</span><span class="cn">TRUE</span>)</span>
<span id="cb62-7"><a href="http://localhost:4845/#cb62-7" aria-hidden="true" tabindex="-1"></a><span class="fu">predict</span>(PCA_Results,<span class="at">newdata =</span> data)[,<span class="dv">1</span>] <span class="sc">*</span> <span class="fu">sign</span>(<span class="fu">sum</span>(PCA_Results<span class="sc">$</span>rotation[,<span class="dv">1</span>]))</span>
<span id="cb62-8"><a href="http://localhost:4845/#cb62-8" aria-hidden="true" tabindex="-1"></a> }</span>
<span id="cb62-9"><a href="http://localhost:4845/#cb62-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-10"><a href="http://localhost:4845/#cb62-10" aria-hidden="true" tabindex="-1"></a><span class="do">### Ensure correlation of higher TMB, FGA, Gleason with more aggressive</span></span>
<span id="cb62-11"><a href="http://localhost:4845/#cb62-11" aria-hidden="true" tabindex="-1"></a><span class="do">##If first PCA is going same direction as gleason, nothing will change. If going opposite direction - becomes negative</span></span>
<span id="cb62-12"><a href="http://localhost:4845/#cb62-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-13"><a href="http://localhost:4845/#cb62-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Use replicate to apply PCA 1 Function many times to dataset  </span></span>
<span id="cb62-14"><a href="http://localhost:4845/#cb62-14" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">13</span>) <span class="do">## Can be any #</span></span>
<span id="cb62-15"><a href="http://localhost:4845/#cb62-15" aria-hidden="true" tabindex="-1"></a>BootstrappedPCA<span class="ot">=</span><span class="fu">replicate</span>(<span class="dv">5000</span>,<span class="fu">Get_Principal_Component_One</span>(Clinical_Train)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="plotting-sample-of-bootstrapped-analysis" class="level2">
<h2 class="anchored" data-anchor-id="plotting-sample-of-bootstrapped-analysis">Plotting Sample of Bootstrapped Analysis<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#plotting-sample-of-bootstrapped-analysis" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li><p>Showing histogram distribution of aggressiveness score (N=9 Patient sample)</p></li>
<li><p>Determining whether higher aggressiveness score corresponds with higher Gleason, TMB, FGA scores</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="http://localhost:4845/#cb63-1" aria-hidden="true" tabindex="-1"></a>BootstrappedPCA <span class="sc">|&gt;</span></span>
<span id="cb63-2"><a href="http://localhost:4845/#cb63-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb63-3"><a href="http://localhost:4845/#cb63-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">set_colnames</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(BootstrappedPCA)) <span class="sc">|&gt;</span></span>
<span id="cb63-4"><a href="http://localhost:4845/#cb63-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Observation =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>(), <span class="at">.before=</span><span class="dv">1</span>) <span class="sc">|&gt;</span></span>
<span id="cb63-5"><a href="http://localhost:4845/#cb63-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>Observation,</span>
<span id="cb63-6"><a href="http://localhost:4845/#cb63-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">'Bootstrap.Iteration'</span>,</span>
<span id="cb63-7"><a href="http://localhost:4845/#cb63-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">'PC1'</span>) <span class="sc">|&gt;</span></span>
<span id="cb63-8"><a href="http://localhost:4845/#cb63-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Observation <span class="sc">%in%</span> <span class="fu">round</span>(<span class="fu">seq</span>(<span class="dv">1</span>, <span class="dv">110</span>, <span class="at">length.out=</span><span class="dv">9</span>))) <span class="sc">|&gt;</span></span>
<span id="cb63-9"><a href="http://localhost:4845/#cb63-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> PC1)) <span class="sc">+</span></span>
<span id="cb63-10"><a href="http://localhost:4845/#cb63-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>() <span class="sc">+</span></span>
<span id="cb63-11"><a href="http://localhost:4845/#cb63-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Observation, <span class="at">nrow=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb63-12"><a href="http://localhost:4845/#cb63-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb63-13"><a href="http://localhost:4845/#cb63-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">'Bootstrapped sampling distribution of aggressiveness scores'</span>,</span>
<span id="cb63-14"><a href="http://localhost:4845/#cb63-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">subtitle =</span> <span class="st">'9 individuals visualized, 10000 bootstrap replications of PCA'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>`stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="http://localhost:4845/#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="do">## 0 represents avg aggress. Computed from Z scores. 2 is more aggressive. Negative repres lower, positive higher. </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="fitting-and-building-the-models" class="level2">
<h2 class="anchored" data-anchor-id="fitting-and-building-the-models">Fitting and Building the Models<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#fitting-and-building-the-models" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li><p>Evaluating whether Use TMRPSS2-ERG fusion status can predict aggressiveness of disease</p></li>
<li><p>Models used</p>
<ol type="1">
<li><p>Linear Regression</p></li>
<li><p>Lasso</p></li>
<li><p>Random Forest</p></li>
</ol></li>
</ul>
</section>
<section id="splitting-the-data" class="level2">
<h2 class="anchored" data-anchor-id="splitting-the-data">Splitting the Data<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#splitting-the-data" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li><p>Splitting the data – allows us to evaluate unbiased accuracy of models</p></li>
<li><p>Created new column called Aggregate Score (Response variable)</p></li>
<li><p>This allows us to make predictions (Clinical Train vs.&nbsp;Clinical Test)</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="http://localhost:4845/#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Need to include PCA Scores we computed in dataframe </span></span>
<span id="cb66-2"><a href="http://localhost:4845/#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Split the data -- obtain unbiased accuracy of models</span></span>
<span id="cb66-3"><a href="http://localhost:4845/#cb66-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-4"><a href="http://localhost:4845/#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="do">## Adds new column called Agg Score (response variable)</span></span>
<span id="cb66-5"><a href="http://localhost:4845/#cb66-5" aria-hidden="true" tabindex="-1"></a>Clinical_Train<span class="sc">$</span>Aggressiveness_Score<span class="ot">=</span>PCA_Results<span class="sc">$</span>x[,<span class="dv">1</span>] <span class="sc">*</span> <span class="fu">sign</span>(<span class="fu">sum</span>(PCA_Results<span class="sc">$</span>rotation[,<span class="dv">1</span>]))</span>
<span id="cb66-6"><a href="http://localhost:4845/#cb66-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-7"><a href="http://localhost:4845/#cb66-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Enables predictions </span></span>
<span id="cb66-8"><a href="http://localhost:4845/#cb66-8" aria-hidden="true" tabindex="-1"></a>Clinical_Test<span class="sc">$</span>Aggressiveness_Score<span class="ot">=</span><span class="fu">predict</span>(PCA_Results, <span class="at">newdata=</span>Clinical_Test)[,<span class="dv">1</span>] <span class="sc">*</span> <span class="fu">sign</span>(<span class="fu">sum</span>(PCA_Results<span class="sc">$</span>rotation[,<span class="dv">1</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="linear-model-1" class="level2">
<h2 class="anchored" data-anchor-id="linear-model-1">Linear Model 1<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#linear-model-1" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li>Uses TMRPSS2-ERG fusion status (binary variable) and Age (continuous variable) to predict aggressiveness of disease</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="http://localhost:4845/#cb67-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Simple LINEAR MODEL </span></span>
<span id="cb67-2"><a href="http://localhost:4845/#cb67-2" aria-hidden="true" tabindex="-1"></a><span class="do">## Use mutation to test aggress score. Uses  age as control</span></span>
<span id="cb67-3"><a href="http://localhost:4845/#cb67-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb67-4"><a href="http://localhost:4845/#cb67-4" aria-hidden="true" tabindex="-1"></a>Simple_OLS <span class="ot">&lt;-</span> <span class="fu">lm</span>(Aggressiveness_Score <span class="sc">~</span> TMPRSS2.Simplified <span class="sc">+</span> Age, Clinical_Train)</span>
<span id="cb67-5"><a href="http://localhost:4845/#cb67-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(Simple_OLS) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = Aggressiveness_Score ~ TMPRSS2.Simplified + Age, 
    data = Clinical_Train)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.89526 -0.81695 -0.03613  0.66585  2.86520 

Coefficients:
                           Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                -4.18172    1.11287  -3.758 0.000312 ***
TMPRSS2.SimplifiedPositive -0.01396    0.24907  -0.056 0.955429    
Age                         0.06698    0.01744   3.840 0.000234 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.172 on 86 degrees of freedom
Multiple R-squared:  0.1471,    Adjusted R-squared:  0.1272 
F-statistic: 7.414 on 2 and 86 DF,  p-value: 0.00107</code></pre>
</div>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="http://localhost:4845/#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Simple_OLS)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-18-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-18-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-18-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="linear-model-2" class="level2">
<h2 class="anchored" data-anchor-id="linear-model-2">Linear Model 2<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#linear-model-2" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li><p>Uses TMRPSS2-ERG fusion status (binary variable) and Age (continuous variable) to predict aggressiveness of disease</p></li>
<li><p>Introduced “*” to the formula to gauge interaction between variables.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="http://localhost:4845/#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Simple LINEAR REGRESSION MODEL + INTERACTION BTWEN TMPRSS and AGE</span></span>
<span id="cb70-2"><a href="http://localhost:4845/#cb70-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-3"><a href="http://localhost:4845/#cb70-3" aria-hidden="true" tabindex="-1"></a>Simple_OLS_Interaction <span class="ot">&lt;-</span> <span class="fu">lm</span>(Aggressiveness_Score <span class="sc">~</span> TMPRSS2.Simplified <span class="sc">*</span> Age, Clinical_Train) </span>
<span id="cb70-4"><a href="http://localhost:4845/#cb70-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(Simple_OLS_Interaction) <span class="do">## Showing that neither Age nor Mutation status are statistically significant predictors of aggressivness score when using OLS model</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = Aggressiveness_Score ~ TMPRSS2.Simplified * Age, 
    data = Clinical_Train)

Residuals:
     Min       1Q   Median       3Q      Max 
-2.85977 -0.76719 -0.06796  0.64914  2.88643 

Coefficients:
                               Estimate Std. Error t value Pr(&gt;|t|)  
(Intercept)                    -4.63082    1.84404  -2.511   0.0139 *
TMPRSS2.SimplifiedPositive      0.68657    2.30021   0.298   0.7661  
Age                             0.07411    0.02914   2.543   0.0128 *
TMPRSS2.SimplifiedPositive:Age -0.01118    0.03649  -0.306   0.7601  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.178 on 85 degrees of freedom
Multiple R-squared:  0.148, Adjusted R-squared:  0.1179 
F-statistic: 4.922 on 3 and 85 DF,  p-value: 0.003353</code></pre>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="http://localhost:4845/#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Simple_OLS_Interaction)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-19-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-19-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-19-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="linear-model-3" class="level2">
<h2 class="anchored" data-anchor-id="linear-model-3">Linear Model 3<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#linear-model-3" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li>Uses TMRPSS2-ERG fusion status (binary variable) and Age (continuous variable) and Mutation Count (continuous) to predict aggressiveness of disease (Added another variable - Mutation Count)</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="http://localhost:4845/#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Complex LINEAR REGRESSION MODEL - evaluates age and mutation account (two variables)</span></span>
<span id="cb73-2"><a href="http://localhost:4845/#cb73-2" aria-hidden="true" tabindex="-1"></a>Complex_OLS_Interaction <span class="ot">&lt;-</span> <span class="fu">lm</span>(Aggressiveness_Score <span class="sc">~</span> TMPRSS2.Simplified <span class="sc">+</span> Age <span class="sc">+</span> Mutation.Count, Clinical_Train)  </span>
<span id="cb73-3"><a href="http://localhost:4845/#cb73-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(Complex_OLS_Interaction) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = Aggressiveness_Score ~ TMPRSS2.Simplified + Age + 
    Mutation.Count, data = Clinical_Train)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.9233 -0.4455 -0.0786  0.4657  1.7375 

Coefficients:
                            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                -3.540141   0.684609  -5.171 1.53e-06 ***
TMPRSS2.SimplifiedPositive -0.337310   0.155116  -2.175   0.0324 *  
Age                         0.019233   0.011415   1.685   0.0957 .  
Mutation.Count              0.077056   0.006429  11.985  &lt; 2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.7189 on 85 degrees of freedom
Multiple R-squared:  0.6829,    Adjusted R-squared:  0.6717 
F-statistic: 61.02 on 3 and 85 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="http://localhost:4845/#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Complex_OLS_Interaction)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-20-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-20-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-20-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="linear-model-4" class="level2">
<h2 class="anchored" data-anchor-id="linear-model-4">Linear Model 4<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#linear-model-4" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li><p>Uses TMRPSS2-ERG fusion status (binary variable) * Age (continuous variable) and Mutation Count (continuous) to predict aggressiveness of disease</p>
<ul>
<li>Added another variable - Mutation Count and introduced “*” to the formula to gauge interaction between variables</li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="http://localhost:4845/#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Complex LINEAR REGRESSION MODEL + INTERACTION BTWEN TMPRSS and AGE</span></span>
<span id="cb76-2"><a href="http://localhost:4845/#cb76-2" aria-hidden="true" tabindex="-1"></a>Complex_OLS_Interaction <span class="ot">&lt;-</span> <span class="fu">lm</span>(Aggressiveness_Score <span class="sc">~</span> TMPRSS2.Simplified <span class="sc">*</span> Age <span class="sc">+</span> Mutation.Count, Clinical_Train)  </span>
<span id="cb76-3"><a href="http://localhost:4845/#cb76-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(Complex_OLS_Interaction) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = Aggressiveness_Score ~ TMPRSS2.Simplified * Age + 
    Mutation.Count, data = Clinical_Train)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.87445 -0.44447 -0.06793  0.43489  1.74172 

Coefficients:
                               Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                    -4.14741    1.12928  -3.673 0.000421 ***
TMPRSS2.SimplifiedPositive      0.61049    1.40775   0.434 0.665645    
Age                             0.02884    0.01823   1.582 0.117403    
Mutation.Count                  0.07712    0.00645  11.956  &lt; 2e-16 ***
TMPRSS2.SimplifiedPositive:Age -0.01513    0.02233  -0.677 0.499999    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.7212 on 84 degrees of freedom
Multiple R-squared:  0.6846,    Adjusted R-squared:  0.6696 
F-statistic: 45.59 on 4 and 84 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="http://localhost:4845/#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Complex_OLS_Interaction)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-21-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-21-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-21-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="linear-model-5" class="level2">
<h2 class="anchored" data-anchor-id="linear-model-5">Linear Model 5<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#linear-model-5" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li>Uses TMRPSS2-ERG fusion status (binary variable) AND Age (continuous variable) and Mutation Count (continuous) AND Capra Score to predict aggressiveness of disease</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="http://localhost:4845/#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Complex LINEAR REGRESSION MODEL + INTERACTION BTWEN TMPRSS and AGE and Capra Score</span></span>
<span id="cb79-2"><a href="http://localhost:4845/#cb79-2" aria-hidden="true" tabindex="-1"></a>PrefOLS_NoTMPRSS2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(Aggressiveness_Score <span class="sc">~</span> Age <span class="sc">+</span> Mutation.Count <span class="sc">+</span> Capra.Score, Clinical_Train)  </span>
<span id="cb79-3"><a href="http://localhost:4845/#cb79-3" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(PrefOLS_NoTMPRSS2) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = Aggressiveness_Score ~ Age + Mutation.Count + Capra.Score, 
    data = Clinical_Train)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.84181 -0.47278 -0.01608  0.41482  1.67989 

Coefficients:
               Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)    -3.76862    0.68997  -5.462  4.6e-07 ***
Age             0.02002    0.01182   1.693   0.0941 .  
Mutation.Count  0.07318    0.00667  10.971  &lt; 2e-16 ***
Capra.Score     0.03988    0.04435   0.899   0.3711    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.7351 on 85 degrees of freedom
Multiple R-squared:  0.6684,    Adjusted R-squared:  0.6567 
F-statistic: 57.12 on 3 and 85 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="http://localhost:4845/#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(PrefOLS_NoTMPRSS2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-22-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-22-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-22-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="lasso-regression" class="level2">
<h2 class="anchored" data-anchor-id="lasso-regression"><strong>Lasso Regression</strong><a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#lasso-regression" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li><p>Linear Regression models indicate that not all predictors are relevant.</p>
<ul>
<li><p>Lasso shrinks some coefficients to zero, effectively removing irrelevant predictors.</p></li>
<li><p>Creates simpler, more interpretable models by focusing only on the most important predictors.</p></li>
</ul></li>
<li><p>Introducing LASSO to remove non-relevant predictors and overcome any over fitting concerns</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="http://localhost:4845/#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="do">## LASSO REGRESSION MODEL</span></span>
<span id="cb82-2"><a href="http://localhost:4845/#cb82-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> TMPRSS2.Simplified <span class="sc">+</span> Interstitial.Deletion <span class="sc">+</span></span>
<span id="cb82-3"><a href="http://localhost:4845/#cb82-3" aria-hidden="true" tabindex="-1"></a>                                  Age <span class="sc">+</span> Mutation.Count <span class="sc">+</span> PSA.Status <span class="sc">+</span></span>
<span id="cb82-4"><a href="http://localhost:4845/#cb82-4" aria-hidden="true" tabindex="-1"></a>                                  Capra.Score <span class="sc">+</span> Tumor.Stage.Binary,</span>
<span id="cb82-5"><a href="http://localhost:4845/#cb82-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> Clinical_Train)[,<span class="sc">-</span><span class="dv">1</span>] <span class="sc">|&gt;</span></span>
<span id="cb82-6"><a href="http://localhost:4845/#cb82-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale</span>() <span class="sc">|&gt;</span> <span class="fu">as.matrix</span>()</span>
<span id="cb82-7"><a href="http://localhost:4845/#cb82-7" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> Clinical_Train<span class="sc">$</span>Aggressiveness_Score</span>
<span id="cb82-8"><a href="http://localhost:4845/#cb82-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-9"><a href="http://localhost:4845/#cb82-9" aria-hidden="true" tabindex="-1"></a><span class="do">## Fit Lasso with cross-validation</span></span>
<span id="cb82-10"><a href="http://localhost:4845/#cb82-10" aria-hidden="true" tabindex="-1"></a>lasso_model <span class="ot">&lt;-</span> <span class="fu">cv.glmnet</span>(x, y, <span class="at">alpha =</span> <span class="dv">1</span>)</span>
<span id="cb82-11"><a href="http://localhost:4845/#cb82-11" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(lasso_model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:  cv.glmnet(x = x, y = y, alpha = 1) 

Measure: Mean-Squared Error 

    Lambda Index Measure     SE Nonzero
min 0.0467    34  0.5866 0.1021       5
1se 0.3296    13  0.6794 0.1255       1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="http://localhost:4845/#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Get coefficients at optimal lambda</span></span>
<span id="cb84-2"><a href="http://localhost:4845/#cb84-2" aria-hidden="true" tabindex="-1"></a>best_lambda <span class="ot">&lt;-</span> lasso_model<span class="sc">$</span>lambda.min</span>
<span id="cb84-3"><a href="http://localhost:4845/#cb84-3" aria-hidden="true" tabindex="-1"></a>lasso_coefficients <span class="ot">&lt;-</span> <span class="fu">coef</span>(lasso_model, <span class="at">s =</span> best_lambda)</span>
<span id="cb84-4"><a href="http://localhost:4845/#cb84-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(lasso_coefficients)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>8 x 1 sparse Matrix of class "dgCMatrix"
                                      s1
(Intercept)                -2.430028e-16
TMPRSS2.SimplifiedPositive -1.378849e-01
Interstitial.DeletionYes    3.091469e-02
Age                         1.019506e-01
Mutation.Count              9.359645e-01
PSA.Status                  .           
Capra.Score                 4.395810e-02
Tumor.Stage.BinarypT3       .           </code></pre>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="http://localhost:4845/#cb86-1" aria-hidden="true" tabindex="-1"></a>lasso_preferred_predictors <span class="ot">=</span> <span class="fu">c</span>(<span class="st">'TMPRSS2.Full'</span>,</span>
<span id="cb86-2"><a href="http://localhost:4845/#cb86-2" aria-hidden="true" tabindex="-1"></a>                               <span class="st">'Age'</span>,</span>
<span id="cb86-3"><a href="http://localhost:4845/#cb86-3" aria-hidden="true" tabindex="-1"></a>                               <span class="st">'Mutation.Count'</span>,</span>
<span id="cb86-4"><a href="http://localhost:4845/#cb86-4" aria-hidden="true" tabindex="-1"></a>                               <span class="st">'Capra.Score'</span>)</span>
<span id="cb86-5"><a href="http://localhost:4845/#cb86-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-6"><a href="http://localhost:4845/#cb86-6" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot cross-validation results</span></span>
<span id="cb86-7"><a href="http://localhost:4845/#cb86-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(lasso_model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="random-forest-1" class="level2">
<h2 class="anchored" data-anchor-id="random-forest-1"><strong>Random Forest 1</strong><a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#random-forest-1" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li>Uses TMRPSS2-ERG fusion status (binary variable) and Age (continuous variable) to predict aggressiveness of disease</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="http://localhost:4845/#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Simple Random Forest Model</span></span>
<span id="cb87-2"><a href="http://localhost:4845/#cb87-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">12509</span>)</span>
<span id="cb87-3"><a href="http://localhost:4845/#cb87-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-4"><a href="http://localhost:4845/#cb87-4" aria-hidden="true" tabindex="-1"></a>Simple_RF <span class="ot">&lt;-</span> <span class="fu">randomForest</span>(Aggressiveness_Score <span class="sc">~</span> TMPRSS2.Simplified <span class="sc">+</span> Age, <span class="at">data =</span> Clinical_Train)</span>
<span id="cb87-5"><a href="http://localhost:4845/#cb87-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Simple_RF)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="http://localhost:4845/#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">importance</span>(Simple_RF)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   IncNodePurity
TMPRSS2.Simplified      3.497077
Age                    37.904188</code></pre>
</div>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="http://localhost:4845/#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">varImpPlot</span>(Simple_RF)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-24-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="random-forest-2" class="level2">
<h2 class="anchored" data-anchor-id="random-forest-2"><strong>Random Forest 2</strong><a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#random-forest-2" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li>Uses TMRPSS2-ERG fusion status (binary variable) and Age (continuous variable) to predict aggressiveness of disease</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="http://localhost:4845/#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Complex Random Forest Model</span></span>
<span id="cb91-2"><a href="http://localhost:4845/#cb91-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">12509</span>)</span>
<span id="cb91-3"><a href="http://localhost:4845/#cb91-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-4"><a href="http://localhost:4845/#cb91-4" aria-hidden="true" tabindex="-1"></a>Complex_RF <span class="ot">&lt;-</span> <span class="fu">randomForest</span>(Aggressiveness_Score <span class="sc">~</span> TMPRSS2.Simplified <span class="sc">+</span> Age <span class="sc">+</span> Mutation.Count, <span class="at">data =</span> Clinical_Train)</span>
<span id="cb91-5"><a href="http://localhost:4845/#cb91-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(Complex_RF)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
 randomForest(formula = Aggressiveness_Score ~ TMPRSS2.Simplified +      Age + Mutation.Count, data = Clinical_Train) 
               Type of random forest: regression
                     Number of trees: 500
No. of variables tried at each split: 1

          Mean of squared residuals: 0.6485356
                    % Var explained: 58.33</code></pre>
</div>
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="http://localhost:4845/#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Complex_RF)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="http://localhost:4845/#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">importance</span>(Complex_RF)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   IncNodePurity
TMPRSS2.Simplified      2.378917
Age                    22.752347
Mutation.Count         66.425482</code></pre>
</div>
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="http://localhost:4845/#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="fu">varImpPlot</span>(Complex_RF)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-25-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="random-forest-3" class="level2">
<h2 class="anchored" data-anchor-id="random-forest-3"><strong>Random Forest 3</strong><a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#random-forest-3" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li>Complex Random Forest + Lasso: this is a Random Forest regression model, built using a set of predictors identified as being preferred through lasso</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="http://localhost:4845/#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Complex Random Forest Model + Lasso Preferred Predictors</span></span>
<span id="cb97-2"><a href="http://localhost:4845/#cb97-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">12509</span>)</span>
<span id="cb97-3"><a href="http://localhost:4845/#cb97-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-4"><a href="http://localhost:4845/#cb97-4" aria-hidden="true" tabindex="-1"></a>lasso_preferred_rf <span class="ot">&lt;-</span> <span class="fu">randomForest</span>(Aggressiveness_Score <span class="sc">~</span> ., <span class="at">data =</span> Clinical_Train[,<span class="fu">c</span>(<span class="st">'Aggressiveness_Score'</span>, lasso_preferred_predictors)])</span>
<span id="cb97-5"><a href="http://localhost:4845/#cb97-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(lasso_preferred_rf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
 randomForest(formula = Aggressiveness_Score ~ ., data = Clinical_Train[,      c("Aggressiveness_Score", lasso_preferred_predictors)]) 
               Type of random forest: regression
                     Number of trees: 500
No. of variables tried at each split: 1

          Mean of squared residuals: 0.716419
                    % Var explained: 53.97</code></pre>
</div>
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="http://localhost:4845/#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(lasso_preferred_rf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-26-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="http://localhost:4845/#cb100-1" aria-hidden="true" tabindex="-1"></a><span class="fu">importance</span>(lasso_preferred_rf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               IncNodePurity
TMPRSS2.Full         5.05445
Age                 22.21267
Mutation.Count      59.81188
Capra.Score         15.13032</code></pre>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="http://localhost:4845/#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">varImpPlot</span>(lasso_preferred_rf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-26-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="linear-regression-model-using-lasso-preferred-predictors" class="level2">
<h2 class="anchored" data-anchor-id="linear-regression-model-using-lasso-preferred-predictors">Linear Regression Model Using Lasso Preferred Predictors<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#linear-regression-model-using-lasso-preferred-predictors" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li>Re-assessed accuracy of Linear Regression Models - took Lasso Preferred Predictors and applied to LR Model</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="http://localhost:4845/#cb103-1" aria-hidden="true" tabindex="-1"></a>lasso_preferred_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(Aggressiveness_Score <span class="sc">~</span> ., <span class="at">data =</span> Clinical_Train[, <span class="fu">c</span>(<span class="st">'Aggressiveness_Score'</span>, lasso_preferred_predictors)])</span>
<span id="cb103-2"><a href="http://localhost:4845/#cb103-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(lasso_preferred_lm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = Aggressiveness_Score ~ ., data = Clinical_Train[, 
    c("Aggressiveness_Score", lasso_preferred_predictors)])

Residuals:
     Min       1Q   Median       3Q      Max 
-1.96751 -0.52198 -0.06846  0.44696  1.61099 

Coefficients:
                                                 Estimate Std. Error t value
(Intercept)                                     -3.586085   0.681061  -5.265
TMPRSS2.FullPositive, no interstitial deletion  -0.565097   0.202420  -2.792
TMPRSS2.FullPositive with interstitial deletion -0.191268   0.174907  -1.094
Age                                              0.018455   0.011534   1.600
Mutation.Count                                   0.076235   0.006559  11.624
Capra.Score                                      0.036902   0.042931   0.860
                                                Pr(&gt;|t|)    
(Intercept)                                     1.08e-06 ***
TMPRSS2.FullPositive, no interstitial deletion   0.00651 ** 
TMPRSS2.FullPositive with interstitial deletion  0.27732    
Age                                              0.11339    
Mutation.Count                                   &lt; 2e-16 ***
Capra.Score                                      0.39251    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.7112 on 83 degrees of freedom
Multiple R-squared:  0.6969,    Adjusted R-squared:  0.6787 
F-statistic: 38.17 on 5 and 83 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="http://localhost:4845/#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(lasso_preferred_lm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-27-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-27-3.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-27-4.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="generate-predictions-for-models" class="level2">
<h2 class="anchored" data-anchor-id="generate-predictions-for-models">Generate Predictions for Models<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#generate-predictions-for-models" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h2>
<ul>
<li>Compute prediction performance for each of the data frames mentioned above
<ul>
<li>Useful metrics are RMSE, MAE, and RSq</li>
<li>Performing on training and testing data (separate data frames)</li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="http://localhost:4845/#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Generate training predictions</span></span>
<span id="cb106-2"><a href="http://localhost:4845/#cb106-2" aria-hidden="true" tabindex="-1"></a>Train_Predictions <span class="ot">=</span> Clinical_Train <span class="sc">|&gt;</span></span>
<span id="cb106-3"><a href="http://localhost:4845/#cb106-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Aggressiveness_Score) <span class="sc">|&gt;</span></span>
<span id="cb106-4"><a href="http://localhost:4845/#cb106-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">LASSO_Pref_LM_No_TMPRSS2 =</span> <span class="fu">predict</span>(PrefOLS_NoTMPRSS2, <span class="at">newdata =</span> Clinical_Train),</span>
<span id="cb106-5"><a href="http://localhost:4845/#cb106-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">LASSO_Pref_LM_With_TMPRSS2 =</span> <span class="fu">predict</span>(lasso_preferred_lm, <span class="at">newdata =</span> Clinical_Train),</span>
<span id="cb106-6"><a href="http://localhost:4845/#cb106-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">Simple_Random_Forest =</span> <span class="fu">predict</span>(Simple_RF, <span class="at">newdata =</span> Clinical_Train),</span>
<span id="cb106-7"><a href="http://localhost:4845/#cb106-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">Full_Random_Forest =</span> <span class="fu">predict</span>(Complex_RF, <span class="at">newdata =</span> Clinical_Train),</span>
<span id="cb106-8"><a href="http://localhost:4845/#cb106-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">LASSO_Pref_Random_Forest =</span> <span class="fu">predict</span>(lasso_preferred_rf, <span class="at">newdata =</span> Clinical_Train))</span>
<span id="cb106-9"><a href="http://localhost:4845/#cb106-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-10"><a href="http://localhost:4845/#cb106-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Generate training metrics</span></span>
<span id="cb106-11"><a href="http://localhost:4845/#cb106-11" aria-hidden="true" tabindex="-1"></a>Train_Metrics <span class="ot">=</span> Train_Predictions <span class="sc">|&gt;</span></span>
<span id="cb106-12"><a href="http://localhost:4845/#cb106-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Aggressiveness_Score, <span class="at">names_to=</span><span class="st">'Model'</span>, <span class="at">values_to=</span><span class="st">'Prediction'</span>) <span class="sc">|&gt;</span></span>
<span id="cb106-13"><a href="http://localhost:4845/#cb106-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Model) <span class="sc">|&gt;</span></span>
<span id="cb106-14"><a href="http://localhost:4845/#cb106-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Train_RMSE =</span> <span class="fu">sqrt</span>(<span class="fu">mean</span>((Aggressiveness_Score<span class="sc">-</span>Prediction)<span class="sc">^</span><span class="dv">2</span>)),</span>
<span id="cb106-15"><a href="http://localhost:4845/#cb106-15" aria-hidden="true" tabindex="-1"></a>            <span class="at">Train_MAE =</span> <span class="fu">mean</span>(<span class="fu">abs</span>(Aggressiveness_Score<span class="sc">-</span>Prediction)),</span>
<span id="cb106-16"><a href="http://localhost:4845/#cb106-16" aria-hidden="true" tabindex="-1"></a>            <span class="at">Train_RSq =</span> <span class="fu">cor</span>(Aggressiveness_Score, Prediction)<span class="sc">^</span><span class="dv">2</span>) <span class="sc">|&gt;</span></span>
<span id="cb106-17"><a href="http://localhost:4845/#cb106-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Train_RMSE)</span>
<span id="cb106-18"><a href="http://localhost:4845/#cb106-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-19"><a href="http://localhost:4845/#cb106-19" aria-hidden="true" tabindex="-1"></a><span class="do">## Generate testing predictions</span></span>
<span id="cb106-20"><a href="http://localhost:4845/#cb106-20" aria-hidden="true" tabindex="-1"></a>Test_Predictions <span class="ot">=</span> Clinical_Test <span class="sc">|&gt;</span></span>
<span id="cb106-21"><a href="http://localhost:4845/#cb106-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Aggressiveness_Score) <span class="sc">|&gt;</span></span>
<span id="cb106-22"><a href="http://localhost:4845/#cb106-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">LASSO_Pref_LM_No_TMPRSS2 =</span> <span class="fu">predict</span>(PrefOLS_NoTMPRSS2, <span class="at">newdata =</span> Clinical_Test),</span>
<span id="cb106-23"><a href="http://localhost:4845/#cb106-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">LASSO_Pref_LM_With_TMPRSS2 =</span> <span class="fu">predict</span>(lasso_preferred_lm, <span class="at">newdata =</span> Clinical_Test),</span>
<span id="cb106-24"><a href="http://localhost:4845/#cb106-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">Simple_Random_Forest =</span> <span class="fu">predict</span>(Simple_RF, <span class="at">newdata =</span> Clinical_Test),</span>
<span id="cb106-25"><a href="http://localhost:4845/#cb106-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">Full_Random_Forest =</span> <span class="fu">predict</span>(Complex_RF, <span class="at">newdata =</span> Clinical_Test),</span>
<span id="cb106-26"><a href="http://localhost:4845/#cb106-26" aria-hidden="true" tabindex="-1"></a>         <span class="at">LASSO_Pref_Random_Forest =</span> <span class="fu">predict</span>(lasso_preferred_rf, <span class="at">newdata =</span> Clinical_Test))</span>
<span id="cb106-27"><a href="http://localhost:4845/#cb106-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-28"><a href="http://localhost:4845/#cb106-28" aria-hidden="true" tabindex="-1"></a><span class="do">## Generate testing metrics</span></span>
<span id="cb106-29"><a href="http://localhost:4845/#cb106-29" aria-hidden="true" tabindex="-1"></a>Test_Metrics <span class="ot">=</span> Test_Predictions <span class="sc">|&gt;</span></span>
<span id="cb106-30"><a href="http://localhost:4845/#cb106-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Aggressiveness_Score, <span class="at">names_to=</span><span class="st">'Model'</span>, <span class="at">values_to=</span><span class="st">'Prediction'</span>) <span class="sc">|&gt;</span></span>
<span id="cb106-31"><a href="http://localhost:4845/#cb106-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Model) <span class="sc">|&gt;</span></span>
<span id="cb106-32"><a href="http://localhost:4845/#cb106-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Test_RMSE =</span> <span class="fu">sqrt</span>(<span class="fu">mean</span>((Aggressiveness_Score<span class="sc">-</span>Prediction)<span class="sc">^</span><span class="dv">2</span>)),</span>
<span id="cb106-33"><a href="http://localhost:4845/#cb106-33" aria-hidden="true" tabindex="-1"></a>            <span class="at">Test_MAE =</span> <span class="fu">mean</span>(<span class="fu">abs</span>(Aggressiveness_Score<span class="sc">-</span>Prediction)),</span>
<span id="cb106-34"><a href="http://localhost:4845/#cb106-34" aria-hidden="true" tabindex="-1"></a>            <span class="at">Test_RSq =</span> <span class="fu">cor</span>(Aggressiveness_Score, Prediction)<span class="sc">^</span><span class="dv">2</span>) <span class="sc">|&gt;</span></span>
<span id="cb106-35"><a href="http://localhost:4845/#cb106-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Test_RMSE)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="http://localhost:4845/#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Combine train and test metrics</span></span>
<span id="cb107-2"><a href="http://localhost:4845/#cb107-2" aria-hidden="true" tabindex="-1"></a>(<span class="at">Combined_Metrics =</span> <span class="fu">full_join</span>(Train_Metrics, Test_Metrics, <span class="at">by=</span><span class="st">'Model'</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 5 × 7
  Model               Train_RMSE Train_MAE Train_RSq Test_RMSE Test_MAE Test_RSq
  &lt;chr&gt;                    &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
1 LASSO_Pref_Random_…      0.540     0.428     0.897     0.753    0.615  0.425  
2 Full_Random_Forest       0.598     0.485     0.870     0.739    0.594  0.458  
3 LASSO_Pref_LM_With…      0.687     0.542     0.697     0.637    0.500  0.598  
4 LASSO_Pref_LM_No_T…      0.718     0.561     0.668     0.571    0.421  0.640  
5 Simple_Random_Fore…      1.04      0.826     0.383     1.01     0.785  0.00709</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="http://localhost:4845/#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Make column plot of model performance</span></span>
<span id="cb109-2"><a href="http://localhost:4845/#cb109-2" aria-hidden="true" tabindex="-1"></a>Combined_Metrics <span class="sc">|&gt;</span></span>
<span id="cb109-3"><a href="http://localhost:4845/#cb109-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>Model, <span class="at">names_to =</span> <span class="st">'Metric'</span>, <span class="at">values_to =</span> <span class="st">'Value'</span>) <span class="sc">|&gt;</span></span>
<span id="cb109-4"><a href="http://localhost:4845/#cb109-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(Metric, <span class="fu">c</span>(<span class="st">'Dataset'</span>, <span class="st">'Metric'</span>), <span class="at">sep=</span><span class="st">'_'</span>) <span class="sc">|&gt;</span></span>
<span id="cb109-5"><a href="http://localhost:4845/#cb109-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Metric <span class="sc">!=</span> <span class="st">'RSq'</span>) <span class="sc">|&gt;</span></span>
<span id="cb109-6"><a href="http://localhost:4845/#cb109-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Metric =</span> <span class="fu">factor</span>(Metric, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">'RMSE'</span>, <span class="st">'MAE'</span>)),</span>
<span id="cb109-7"><a href="http://localhost:4845/#cb109-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">Dataset =</span> <span class="fu">factor</span>(Dataset, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">'Train'</span>, <span class="st">'Test'</span>))) <span class="sc">|&gt;</span></span>
<span id="cb109-8"><a href="http://localhost:4845/#cb109-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Dataset, Metric) <span class="sc">|&gt;</span></span>
<span id="cb109-9"><a href="http://localhost:4845/#cb109-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Value) <span class="sc">|&gt;</span></span>
<span id="cb109-10"><a href="http://localhost:4845/#cb109-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Model =</span> <span class="fu">fct_inorder</span>(Model)) <span class="sc">|&gt;</span></span>
<span id="cb109-11"><a href="http://localhost:4845/#cb109-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y=</span><span class="fu">fct_rev</span>(Model), <span class="at">x=</span>Value, <span class="at">fill=</span><span class="fu">fct_rev</span>(Dataset))) <span class="sc">+</span></span>
<span id="cb109-12"><a href="http://localhost:4845/#cb109-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Metric) <span class="sc">+</span></span>
<span id="cb109-13"><a href="http://localhost:4845/#cb109-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position=</span><span class="st">'dodge'</span>) <span class="sc">+</span></span>
<span id="cb109-14"><a href="http://localhost:4845/#cb109-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">'Comparison of Model Predictive Performance'</span>,</span>
<span id="cb109-15"><a href="http://localhost:4845/#cb109-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span><span class="st">''</span>, <span class="at">x =</span><span class="st">''</span>, <span class="at">fill =</span> <span class="st">'Dataset'</span>) <span class="sc">+</span></span>
<span id="cb109-16"><a href="http://localhost:4845/#cb109-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_discrete</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="st">'Train'</span>, <span class="st">'Test'</span>)) <span class="sc">+</span></span>
<span id="cb109-17"><a href="http://localhost:4845/#cb109-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb109-18"><a href="http://localhost:4845/#cb109-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">'top'</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="./MBMI 5030_Final Project_files/unnamed-chunk-30-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="conclusion" class="level1">
<h1>Conclusion</h1>
<section id="overall-finding" class="level3">
<h3 class="anchored" data-anchor-id="overall-finding">Overall Finding<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#overall-finding" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h3>
<p>The overall analysis has proven the hypothesis false. In other words, TMPRSS2-ERG Fusion Status cannot be used to predict aggressiveness of disease, as defined by the composite score that consists of radical prostatectomy gleason score, fraction genome altered (FGA) score, and tumor mutation burden (TBM).</p>
</section>
<section id="summary" class="level3">
<h3 class="anchored" data-anchor-id="summary">Summary<a class="anchorjs-link " aria-label="Anchor" data-anchorjs-icon="" href="http://localhost:4845/#summary" style="font: 1em / 1 anchorjs-icons; margin-left: 0.1875em; padding-right: 0.1875em; padding-left: 0.1875em;"></a></h3>
<p>Model accuracy was primarily ascertained using two key metrics – Root mean square deviation (RMSE) and Mean Absolute Error (MAE). RMSE measures the square root of the average squared differences between predictions and actual values whereas MAE measures the average absolute differences between predictions and actual values. In the final histogram you will see turquoise and red bars where we analyze the train versus test data. In this instance the “Train” dataset represents how well the model fits the data and the “test” dataset evaluates performance on unseen data, indicating (and how well the model generalizes). Within the five final models that were chosen for the final comparison and analysis (1- LASSO Pref RF; 2 - Full RF; 3 - LASSO Pref LM + TMPRSS2, 4 - LASSO Pref LM - TMPRSS2; 5 - Simple RF) we viewed considerable variability in predictive performance. Specifically, Models 1 and 2 demonstrate overfitting . A key model of interest LASSO_Pref_LM_No_TMPRSS2 created a more accurate model, which is especially informative given that the LASSO preferred predictors specifically excluded presence of the TMPRSS2 mutation – which we were initially using to determine whether there was an ability to predict aggressiveness of disease.</p>
<p><strong>Limitations</strong></p>
<p>Despite being deemed sufficient for the purposes of this project, the sample size included in our analysis (n = 112) is modest. This likely led to a degree of reduced statistical power, increased bias risk, and lower generalizability. Furthermore, the dataset was curated in 2012. The key metrics still hold utility for prostate cancer screening, diagnosis, and prognosis, however we now likely have a more nuanced understanding of the disease (from 12 years ago) and there might be more impactful variables that we could have used to generate the compositive score and/or include in our analysis. Finally, I believe that greater longitudinal visibility into the patient journey might have unlocked new insights regarding treatment decision, impact of therapeutic pressure, and ultimately how this linked back to aggressiveness of disease. The current sample appears somewhat heterogeneous in terms of disease state and decision to evaluate post-prostatectomy.</p>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->





<script>window.backupDefine = window.define; window.define = undefined;</script>
<script type="text/javascript" src="./MBMI 5030_Final Project_files/quarto-preview.js.download"></script>
<script>window.define = window.backupDefine; window.backupDefine = undefined;</script>
<script type="text/javascript">
  const options = {
    origin: "",
    search: "",
    inputFile: "C:\Users\evans\Documents\BMIN503_Final_Project\MBMI 5030_Final Project_7 Nov 2024.qmd",
    isPresentation: false
  }
  document.addEventListener("DOMContentLoaded", function () {
    window.QuartoPreview.init(options);
  });
</script>
<div><div dir="ltr" class="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"></div></div><i tabindex="0" role="none" data-tabster-dummy="" aria-hidden="true" style="position: fixed; height: 1px; width: 1px; opacity: 0.001; z-index: -1; content-visibility: hidden; top: 0px; left: 0px;"></i></body></html>